pkg/cmd/codespace/ports.go

```diff
@@ -249,14 +275,49 @@ func (a *App) UpdatePortVisibility(ctx context.Context, codespaceName string, ar
 	}
 	defer safeClose(session, &err)
 
+	errc := make(chan error, 1)
+
 	// TODO: check if port visibility can be updated in parallel instead of sequentially
 	for _, port := range ports {
```

Comment:
Since you are referencing `port` in a goroutine I think you will run into a scoping issue like: https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables

You can either modify your anonymous `go func` to accept a `port` argument like `go func(port portType)` and pass it in as an argument, or you can define a new var `port := port` within the for loop

Reply:
It's good to be aware of this pitfall, but it's not necessary to do anything in this function since the child goroutines don't outlive the loop iteration that creates them.